flow/workflows/sync_flow.go

```diff
@@ -195,7 +169,7 @@ func SyncFlowWorkflow(
 			break
 		}
 
-		restart := currentSyncFlowNum >= maxSyncsPerSyncFlow || syncErr
+		restart := syncErr || workflow.GetInfo(ctx).GetContinueAsNewSuggested()
```

Comment:
need to check how this behaves, if cdc flow keeps going until it restarts we could end up with stuttering where reconnect happens twice within a short period (this restarts, then 10 syncs later cdc restarts)